### DISCUSSION TITLE
Support Assert.Warn? (Similar to NUnit)

### DISCUSSION THREAD

**Discussion by james-zhang-hub** [xunit/xunit#2676] [Ideas]
[NUnit supports Warnings](https://docs.nunit.org/articles/nunit/writing-tests/Warnings.html) which "give a warning message but continue execution."  The tests will complete with a status of "Warning" rather than "Passed" or "Failed".

This feature would be useful for testing issues we want to be made aware of, but not severe enough to fail our testing.

Let me know what you think.

**Comment by bradwilson** [comment]
v3 would be the time to add such a feature, but I think I'd be tempted to support warnings without actually adding an `Assert.Warn` so third party libraries can issue warnings, even though I don't believe that `Assert.Warn` is something we'd want to support for pure unit testing scenarios. Warnings imply that a human being has to monitor the process, which is contrary to our goals for unit testing (firm pass or fail).

**Reply by bradwilson** [reply]
Opened issue #2726
